xhu.py

```diff
@@ -65,12 +76,45 @@ def get_info(path: str, root: pathlib.Path) -> typing.Tuple[
         path,
         pathlib.Path(app.config["DATA_ROOT"]),
     )
-
     data_file, metadata_file = get_paths(dest_path)
-
     return data_file, load_metadata(metadata_file)
 
 
+def apply_quota(root: pathlib.Path, quota: int):
+    """ Get the files, sorted by last modification date and the sum of their
+        sizes.
+    """
+    if not quota:
+        return
+
+    file_list = []
+    total_size = 0
+    # We assume a file structure whereby files are are stored inside
+    # uuid() directories inside the root dir and that there aren't any files in
+    # the root dir itself.
```

Comment:
Do not make assumptions about the directory structure. While there is currently only an implementation of this in Prosody, I don’t see why we should rely on only prosody being used on the other side.

Reply:
>  I don’t see why we should rely on only prosody being used on the other side

I'm using Prosody and I'm trying to get something working and running sooner rather than later.
I understand that you'd rather prefer less coupled and more generic solutions.